scripts/base/frameworks/cluster/main.zeek

```diff
@@ -366,11 +435,12 @@ event Broker::peer_lost(endpoint: Broker::EndpointInfo, msg: string) &priority=1
 			Cluster::log(fmt("node down: %s", node_name));
 			delete n$id;
 
+			local active_ids = active_node_ids[n$node_type];
+			delete active_ids[endpoint$id];
+			active_node_ids[n$node_type] = active_ids;
```

Comment:
At the risk of missing something intricate: Is this "just" to make shorter lines, or did one line not somehow do what it should?
```
delete active_node_ids[n$node_type][endpoint$id]
```

Reply:
> delete active_node_ids[n$node_type][endpoint$id]

This won't do what you expect it to do. Maybe this creates a temp copy of `active_node_ids[n$node_type]`?

Reply:
As in DM, the `delete` here will actually work, it's the `add` part for the same table that caused confusion to the &default usage not assigning (https://github.com/zeek/zeek/discussions/2668)

```
	local active_ids = active_node_ids[n$node_type];
	add active_ids[id];
	active_node_ids[n$node_type] = active_ids;
```

IMO (but IMO only), dropping `&default` and explicitly initializing would be less surprising
```
if ( n$node_type !in active_node_ids )
    active_node_ids[n$node_type] = set();

add active_node_ids[n$node_type][id]
```

Reply:
Yep, I got confused here. I like the explicit initialization. Seems indeed easier to understand.